transcom/mymove
# B 20076 MAIN

## [B-20076](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A950884)
## [B-20075](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A950883)
## [B-20074](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A950882)

## [Integration Review](https://github.com/transcom/mymove/pull/13158)
## [Second Integration Review](https://github.com/transcom/mymove/pull/13220)

## Summary

These tasks were combine due to the changes being so similar and requiring changes to some of the same functions in custom.js

These task were to update the Max Billable weight and Max estimated on the Move Task Order, Payment Request, and Review Weights views for office users. The majority of the changes were updating the Max billable to 110% of the combined estimated weight and the Actual Billable weight to the lowest between 110% of he prime estimated or the actual.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

Setup:
1. Access the milmove application
2. Login as a new Customer
3. Create a new move
4. Add at least one HHG, NTS, and NTSR shipments to the move
5. Submit the move
6. Log in as a Service counselor
7. Find the move you created and Submit the move details
8. Log in as a TOO
9. Find the move you created and Approve all shipments
10. Log is as Prime Sim
11. Locate the move you created
12. Update the shipments with estimated and actual weights (you must choose pickup and dropoff dates in the future otherwise your request to update will be rejected)

B-20076
1. Access the Office app
2. Login as a TOO/TIO
3. Find the move you created
4. Navigate to the Move Task Order tab
5. Verify that the max billable weight is 110% of HHG/NTS/NTSR (use previously recorded weight for ntsr)
6. Verify that actual billable weight does not exceed 110% of HHG/NTS/NTSR (use previously recorded weight for ntsr)

B-20075
1. Access the Office app
2. Login as a TIO
3. Find the move you created
4. Navigate to Payment Requests tab
5. Verify that the max billable weight is 110% of HHG/NTS/NTSR (use previously recorded weight for ntsr)
7. Verify that the NTSR shipment has 110% of the estimated weight below the NTSR name, there is a tooltip to describe what it is

B-20074
1. Access the Office Application
2. Login as a TIO
3. Find the move you created
4. Navigate to the Payment Requests tab
5. Select the Review Weights button
6. Verify that the max billable weight is 110% of HHG/NTS/NTSR (use previously recorded weight for ntsr)
7. Verify that the displayed weight for  HHG/NTS/NTSR is either 110% prime estimated weight (use previously recorded weight for ntsr) or the actual weight, whichever is lower
8. Verify that Actual billable weight is a combination of the shipment weights listed below it
9. Verify that Estimated and Original weights display for NTSR shipments


To test multiple iterations (some with estimated weight higher, some with actual weight higher) you can save time on creation by editing the prime_estimated_weight, prime_actual_weight, and ntsRecordedWeight values in the mto_shipment table for the shipment in question. 

### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
